pkg/cluster/cluster.go

```diff
@@ -1199,10 +1209,10 @@ func (c *Cluster) Delete() error {
 		}
 	}
 
-	if err := c.deletePatroniClusterObjects(); err != nil {
+	if err := c.deletePatroniResources(); err != nil {
```

Comment:
shouldn't we skip it if `!c.patroniUsesKubernetes()`?

Reply:
Good question. In the main branch this code is there to not do "blind" K8s API calls against resources that do not exist in infrastructure. In this PR we only delete based on the cluster struct. Therefore, we should only delete what is also allowed to exist.

Reply:
just to avoid unnecessary code execution:)

Reply:
With this delete code there will be no unnecessary API calls at least. One of either ConfigMaps and Endpoints will always get created - no matter what patroniUsesKubernetes() returns (when called inside patroniKubernetesUseConfigMaps). Yes, without right config this leads to errors during create.